### PR TITLE
Add `--verification-key` option to `committee key-hash` command

### PR DESCRIPTION
**Pull request by newhoggy**
# Changelog

```yaml
- description: |
    Add `--verification-key` option to `committee key-hash` command
# uncomment types applicable to the change:
  type:
  # - feature        # introduces a new feature
  # - breaking       # the API has changed in a breaking way
  - compatible     # the API has changed but is non-breaking
  # - optimisation   # measurable performance improvements
  # - improvement    # QoL changes e.g. refactoring
  # - bugfix         # fixes a defect
  # - test           # fixes/modifies tests
  # - maintenance    # not directly related to the code
  # - release        # related to a new release preparation
  # - documentation  # change in code docs, haddocks...
```

# Context

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages
- [ ] The change log section in the PR description has been filled in
- [ ] New tests are added if needed and existing tests are updated.  These may include:
  - golden tests
  - property tests
  - round trip tests
  - integration tests
  See [Running tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [ ] The version bounds in `.cabal` files are updated
- [ ] CI passes. See note on CI.  The following CI checks are required:
  - [ ] Code is linted with `hlint`.  See `.github/workflows/check-hlint.yml` to get the `hlint` version
  - [ ] Code is formatted with `stylish-haskell`.  See `.github/workflows/stylish-haskell.yml` to get the `stylish-haskell` version
  - [ ] Code builds on Linux, MacOS and Windows for `ghc-8.10.7` and `ghc-9.2.7`
- [ ] The changelog section in the PR is updated to describe the change
- [ ] Self-reviewed the diff

### CODE REVIEW

File: cardano-cli/test/cardano-cli-golden/files/golden/help.cli [hunk 1]
```diff
@@ -794,7 +794,10 @@ Usage: cardano-cli conway governance committee key-gen-hot --verification-key-fi
 
   Create a cold key pair for a Constitutional Committee Member
 
-Usage: cardano-cli conway governance committee key-hash --verification-key-file FILE
+Usage: cardano-cli conway governance committee key-hash 
+                                                          ( --verification-key STRING
```
Comment (newhoggy) [inline cardano-cli/test/cardano-cli-golden/files/golden/help.cli:798]: Adds this command line option ⬆️

File: cardano-cli/test/cardano-cli-golden/files/golden/help/conway_governance_committee_key-hash.cli [hunk 2]
```diff
@@ -1,8 +1,15 @@
...
   Print the identifier (hash) of a public key
 
 Available options:
+  --verification-key STRING
+                           Constitutional Committee Member key (hot or cold)
+                           (Bech32-encoded)
   --verification-key-file FILE
-                           Input filepath of the verification key.
+                           Input filepath of the Constitutional Committee Member
+                           key (hot or cold).
```
Comment (newhoggy) [inline cardano-cli/test/cardano-cli-golden/files/golden/help/conway_governance_committee_key-hash.cli:14]: More descriptive help.

File: cardano-cli/src/Cardano/CLI/EraBased/Run/Governance/Committee.hs [hunk 3]
```diff
@@ -109,16 +114,25 @@ data SomeCommitteeKey f
...
+              [ FromSomeType (AsVerificationKey AsCommitteeHotKey ) ACommitteeHotKey
+              , FromSomeType (AsVerificationKey AsCommitteeColdKey) ACommitteeColdKey
+              ]
+        pure (deserialiseAnyOfFromBech32 asTypes (unAnyVerificationKeyText vkText))
+          & onLeft (left . GovernanceCommitteeCmdKeyDecodeError . InputBech32DecodeError)
+      AnyVerificationKeySourceOfFile vkeyPath -> do
+        let asTypes =
+              [ FromSomeType (AsVerificationKey AsCommitteeHotKey ) ACommitteeHotKey
+              , FromSomeType (AsVerificationKey AsCommitteeColdKey) ACommitteeColdKey
+              ]
```
Comment (newhoggy) [inline cardano-cli/src/Cardano/CLI/EraBased/Run/Governance/Committee.hs:132]: We are still reasonably strict about what keys are accepted.

File: cardano-cli/src/Cardano/CLI/Types/Key/AnyVerificationKeySource.hs [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+  ( AnyVerificationKeySource(..)
+  ) where
+
+import           Cardano.Api
+
+import           Cardano.CLI.Types.Key.AnyVerificationKeyText
+
+-- | The source from which a verification key of an unspecified key role can be
+-- derivied.
+data AnyVerificationKeySource
```
Comment (newhoggy) [inline cardano-cli/src/Cardano/CLI/Types/Key/AnyVerificationKeySource.hs]: Using the naming convention that `XyzSource` is a thing from which an `Xyz` can be derived.

In this case, the `Xyz` is `VerificationKey`.  The `Any` prefix means any `keyrole`.

File: cardano-cli/src/Cardano/CLI/Types/Key/AnyVerificationKeyText.hs
Comment (carbolymer) [inline cardano-cli/src/Cardano/CLI/Types/Key/AnyVerificationKeyText.hs:1]: Does this need to be a separate module? This newtype isn't that big.

File: cardano-cli/cardano-cli.cabal [hunk 5]
```diff
@@ -123,6 +123,8 @@ library
                         Cardano.CLI.Types.Common
                         Cardano.CLI.Types.Governance
                         Cardano.CLI.Types.Key
+                        Cardano.CLI.Types.Key.AnyVerificationKeyText
```
Comment (Jimbo4350) [inline cardano-cli/cardano-cli.cabal]: Separate modules for this is overkill. `Cardano.CLI.Types.Key.VerificationKey` is enough.

**Review by carbolymer — APPROVED** [review]

**Review by Jimbo4350 — APPROVED** [review]

**Pull request merged**